File: src/webui/app/logic/domain/inbox-task-source.ts
import { InboxTaskSource } from "@jupiter/webapi-client";

export function allowUserChanges(source: InboxTaskSource): boolean {
  // Keep synced with python:inbox_task_source.py
  return (
    source === InboxTaskSource.USER ||
    source === InboxTaskSource.BIG_PLAN ||
    source === InboxTaskSource.SLACK_TASK ||
    source === InboxTaskSource.EMAIL_TASK
  );
}

export function inboxTaskSourceName(source: InboxTaskSource): string {
  switch (source) {
    case InboxTaskSource.USER:
      return "User";
    case InboxTaskSource.WORKING_MEM_CLEANUP:
      return "Working Mem Cleanup";
    case InboxTaskSource.HABIT:
      return "Habit";
    case InboxTaskSource.CHORE:
      return "Chore";
    case InboxTaskSource.BIG_PLAN:
      return "Big Plan";
    case InboxTaskSource.JOURNAL:
      return "Journal";
    case InboxTaskSource.METRIC:
      return "Metric";
    case InboxTaskSource.PERSON_CATCH_UP:
      return "Catch Up";
    case InboxTaskSource.PERSON_BIRTHDAY:
      return "Birthday";
    case InboxTaskSource.SLACK_TASK:
      return "Slack";
    case InboxTaskSource.EMAIL_TASK:
      return "Email";
  }
}

File: src/core/jupiter/core/domain/concept/inbox_tasks/inbox_task_source.py
"""The origin of an inbox task."""

from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
class InboxTaskSource(EnumValue):
    """The origin of an inbox task."""

    USER = "user"
    WORKING_MEM_CLEANUP = "working-mem-cleanup"
    HABIT = "habit"
    CHORE = "chore"
    BIG_PLAN = "big-plan"
    JOURNAL = "journal"
    METRIC = "metric"
    PERSON_CATCH_UP = "person-catch-up"
    PERSON_BIRTHDAY = "person-birthday"
    SLACK_TASK = "slack-task"
    EMAIL_TASK = "email-task"

    @property
    def allow_user_changes(self) -> bool:
        """Allow user changes for an inbox task."""
        return self in (
            InboxTaskSource.USER,
            InboxTaskSource.BIG_PLAN,
            InboxTaskSource.SLACK_TASK,
            InboxTaskSource.EMAIL_TASK,
        )
